components/esp_system/esp_ipc.c

```diff
@@ -150,11 +158,12 @@ static esp_err_t esp_ipc_call_and_wait(uint32_t cpu_id, esp_ipc_func_t func, voi
     xSemaphoreTake(s_ipc_mutex[0], portMAX_DELAY);
 #endif
 
-    s_func[cpu_id] = func;
-    s_func_arg[cpu_id] = arg;
-    s_ipc_wait[cpu_id] = wait_for;
-    xSemaphoreGive(s_ipc_sem[cpu_id]);
-    xSemaphoreTake(s_ipc_ack[cpu_id], portMAX_DELAY);
+    atomic_store(&s_func[cpu_id], (uintptr_t)func);
+    atomic_store(&s_func_arg[cpu_id], (uintptr_t)arg);
+    atomic_store(&s_ipc_wait[cpu_id], (int)wait_for);
+    atomic_store(&s_ipc_ack[cpu_id], (uintptr_t)task_handler);
```

Comment:
This `task_handler` may not exist if `CONFIG_ESP_IPC_USES_CALLERS_PRIORITY` is not set

Reply:
Thank you for reporting. Fixed by moving `task_handler` above `CONFIG_ESP_IPC_USES_CALLERS_PRIORITY `.